src/index.ts

```diff
@@ -323,21 +312,29 @@ export default function solidPlugin(options: Partial<Options> = {}): Plugin {
         ? ['solid-js', 'solid-js/web', 'solid-js/store', 'solid-js/html', 'solid-js/h']
         : [];
 
-      const test =
-        userConfig.mode === 'test'
-          ? {
-              test: {
-                globals: true,
-                ...(options.ssr ? {} : { environment: 'jsdom' }),
-                transformMode: {
-                  [options.ssr ? 'ssr' : 'web']: [/\.[jt]sx?$/],
-                },
-                ...(isJestDomInstalled()
-                  ? { setupFiles: [require.resolve('@testing-library/jest-dom/extend-expect.js')] }
-                  : {}),
-              },
-            }
-          : {};
+      const test = (userConfig as any).test || {};
+
+      if (userConfig.mode === 'test') {
+        if (!test.environment && !options.ssr) {
+          test.environment = 'jsdom';
+        }
+        if (!test.transformMode) {
+          test.transformMode = { [options.ssr ? 'ssr' : 'web']: [/\.[jt]sx?$/] };
+        }
+        const jestDomImport = [
+          '@testing-library/jest-dom/vitest',
+          '@testing-library/jest-dom/extend-expect',
+        ].find((path) => {
+          try {
+            require(path);
+          } catch (e) {
+            return false;
+          }
+        });
```

Comment:
This could be decoupled to the function for readability like the old `isJestDomInstalled()`

Reply:
I thought about it, but is it really necessary? it's only called once anyways.

Reply:
I added the ability to detect if this is already in the setupFiles, and moved it to a function.